python/tests/graph/base_test_generators.py

```diff
@@ -0,0 +1,314 @@
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import absolute_import, division, print_function
+
+from collections import namedtuple
+from contextlib import contextmanager
+from glob import glob
+import itertools
+import os
+import shutil
+import tempfile
+
+import numpy as np
+import tensorflow as tf
+
+from sparkdl.graph.input import *
+import sparkdl.graph.utils as tfx
+
+TestCase = namedtuple('TestCase', ['bool_result', 'err_msg'])
+TestFn = namedtuple('TestFn', ['test_fn', 'description'])
+_GinInfo = namedtuple('_GinInfo', ['gin', 'description'])
+
+
+class TestGenBase(object):
+    def __init__(self, vec_size=17, test_batch_size=231):
+        # Testing data spec
+        self.vec_size = vec_size
```

Comment:
this class has an enormous amount of state and moving parts, I cannot keep track what is happening. Please encapsulate what you need to pass to the relevant pieces.